Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -172,3 +188,95 @@ function IsFilePathSpecified
                 [StringComparison]::OrdinalIgnoreCase)
 }
 
+
+function GetVersion($uri, $headers)
+{
+    $version = $null
+
+    Try
+    {
+        $jsonResp = Invoke-RestMethod -Uri $uri -Method Get -Headers $headers
+
+        if ($jsonResp)
+        {
+            $version = $jsonResp.SonarQube.Version
+        }
+
+    }
+    Catch [System.Net.WebException]
+    {
+        Write-Verbose "WebException while trying to invoke $url"
+    }
+
+    return $version
+}
+
+#
+# Helper that returns the version number of the SonarQube server
+#
+function GetSonarQubeServerVersion()
+{
+    param([String][ValidateNotNullOrEmpty()]$serverUrl,
+          [String]$userName,
+          [String]$password)
+
+    Write-Host "Fetching SonarQube server version.."
+
+    $httpHeaders = @{}
+    $serverUri = New-Object -TypeName System.Uri -ArgumentList $serverUrl
+    $serverApiUri = New-Object -TypeName System.Uri -ArgumentList ($serverUri, "/api/system/info")
+
+    $base64Auth = [System.Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes("$($userName):$($password)"))
+    $base64AuthHeader = "Basic $base64Auth"
+
+    $httpHeaders.Add('Authorization', $base64AuthHeader)
+
+    $sqVersion = GetVersion $serverApiUri $httpHeaders
+
+    if(!$sqVersion)
+    {
+        Write-Verbose "Trying to fetch SonarQube version number again.."
+        Start-Sleep -s 2
+
+        $sqVersion = GetVersion $serverApiUri $httpHeaders
+    }
+
+    Write-Verbose "Returning SonarQube server version:$sqVersion"
+    return $sqVersion
+}
+
+#
+# Helper that returns the major version number of the SonarQube server
+#
+function GetSQMajorVersionNumber()
```

Comment:
UT's currently do not exist.. task framework team is working on creating one.. the date for that is tentative end of year I think..